code/lib/core-common/src/utils/validate-config.ts

```diff
@@ -1,19 +1,47 @@
 import { dedent } from 'ts-dedent';
+import { frameworkPackages } from './get-storybook-info';
 
 const renderers = ['html', 'preact', 'react', 'server', 'svelte', 'vue', 'vue3', 'web-components'];
 
 const rendererNames = [...renderers, ...renderers.map((renderer) => `@storybook/${renderer}`)];
 
-// Checks that the framework name is not a renderer
-export function validateFrameworkName(frameworkName: string) {
+export function validateFrameworkName(frameworkName: string | undefined) {
+  const automigrateMessage = `Please run 'npx storybook@next automigrate' to automatically fix your config.
+
+  See the migration guide for more information: 
+  https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#new-framework-api\n`;
+
+  // Throw error if there is no framework field
+  // TODO: maybe this error should not be thrown if we allow empty Storybooks that only use "refs" for composition
+  if (!frameworkName) {
+    throw new Error(dedent`
+      Could not find a 'framework' field in Storybook config.
+
+      ${automigrateMessage}
+    `);
+  }
+
+  // Account for legacy scenario where the framework was referring to a renderer
   if (rendererNames.includes(frameworkName)) {
     throw new Error(dedent`
-      Invalid value of ${frameworkName} in the 'framework' field of Storybook config.
+      Invalid value of '${frameworkName}' in the 'framework' field of Storybook config.
 
-      Please run 'npx sb@next automigrate'
+      ${automigrateMessage}
+    `);
+  }
+
+  // If we know about the framework, we don't need to validate it
+  if (Object.keys(frameworkPackages).includes(frameworkName)) {
+    return;
+  }
+
+  // If it's not a known framework, we need to validate that it's a valid package at least
+  try {
+    require.resolve(frameworkName);
```

Comment:
How about this instead which is more specific, since we're using `frameworkName/preset` below in the code

```
require.resolve(join(frameworkName, 'preset'))
```

@ndelangen points out that this could be problematic in PnP mode, but I think it's OK because the `frameworkName` should be an absolute path in PnP mode anyway. WDYT?

Reply:
Not sure, I’ll wait for Norbert’s response!

Reply:
he agrees with me, we discussed